chain/chain/src/types.rs

```diff
@@ -113,8 +113,8 @@ pub trait RuntimeAdapter: Send + Sync {
 
     fn get_tries(&self) -> ShardTries;
 
-    /// Returns trie.
-    fn get_trie_for_shard(&self, shard_id: ShardId) -> Arc<Trie>;
+    /// Returns trie if shard_id is valid
+    fn get_trie_for_shard(&self, shard_id: ShardId) -> Result<Arc<Trie>, Error>;
```

Comment:
Why create error case for garbage passed to the function?
How would the caller handle this error?
If we have `ShardId`s coming from someone external we should validate them as soon as possible instead of getting an error all the way in here.

Reply:
`store-validator` will fail if there is a wrong ShardChunk stored with incorrectly large `shard_id`. How does invalid `shard_id` case is handled for now? `Vector out of bounds error`?

Reply:
I'm okay to revert it if it's guaranteed somehow that it's impossible to call `get_trie_for_shard` with invalid `shard_id`.

Reply:
>How does invalid shard_id case is handled for now?
Probably in the places where we validate blocks/chunks/headers coming from the network?

If we want to make the code more robust we can make conversion from int to `ShardId` do the validation. By the time we get to `get_trie_for_shard` we shouldn't be dealing with garbage ShardId so in correct code 100% callers will just `unwrap()`.

Reply:
++